includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -342,7 +342,7 @@ protected function read_attributes( &$product ) {
 	protected function read_downloads( &$product ) {
 		$meta_values = array_filter( (array) maybe_unserialize( get_post_meta( $product->get_id(), '_downloadable_files', true ) ) );
 
-		if ( $meta_values ) {
+		if ( ! empty( $meta_values ) && is_array( $meta_values ) ) {
```

Comment:
I think that `is_array` check is redundant, there is cast to array in 343 line. Also `array_filter` accepts only arrays and it returns array